keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -363,13 +388,131 @@ Other components:
+1. CPU: decrease Requests, increase Limits, Memory: decrease Requests, increase Limits.
+
+E2E tests for Burstable class single container Pod that specifies CPU only:
+1. Increase, decrease CPU - Requests only.
+1. Increase, decrease CPU - Limits only.
+1. Increase, decrease CPU - both Requests & Limits.
+
+E2E tests for Burstable class single container Pod that specifies memory only:
+1. Increase, decrease memory - Requests only.
+1. Increase, decrease memory - Limits only.
+1. Increase, decrease memory - both Requests & Limits.
+
+E2E tests for Guaranteed class Pod with three containers (c1, c2, c3):
+1. Increase CPU & memory for all three containers.
+1. Decrease CPU & memory for all three containers.
+1. Increase CPU, decrease memory for all three containers.
+1. Decrease CPU, increase memory for all three containers.
+1. Increase CPU for c1, decrease c2, c3 unchanged - no net CPU change.
+1. Increase memory for c1, decrease c2, c3 unchanged - no net memory change.
+1. Increase CPU for c1, decrease c2 & c3 - net CPU decrease for Pod.
+1. Increase memory for c1, decrease c2 & c3 - net memory decrease for Pod.
+1. Increase CPU for c1 & c3, decrease c2 - net CPU increase for Pod.
+1. Increase memory for c1 & c3, decrease c2 - net memory increase for Pod.
+
+### Resource Quota and Limit Ranges
+
+Setup a namespace with ResourceQuota and a single, valid Pod.
+1. Resize the Pod within resource quota - CPU only.
+1. Resize the Pod within resource quota - memory only.
+1. Resize the Pod within resource quota - both CPU and memory.
+1. Resize the Pod to exceed resource quota - CPU only.
+1. Resize the Pod to exceed resource quota - memory only.
+1. Resize the Pod to exceed resource quota - both CPU and memory.
+
+Setup a namespace with min and max LimitRange and create a single, valid Pod.
+1. Increase, decrease CPU within min/max bounds.
+1. Increase CPU to exceed max value.
+1. Decrease CPU to go below min value.
+1. Increase memory to exceed max value.
+1. Decrease memory to go below min value.
+
+### Resize Policy Tests
+
+Setup a guaranteed class Pod with two containers (c1 & c2).
+1. No resize policy specified, defaults to NoRestart. Verify that CPU and
+   memory are resized without restarting containers.
+1. NoRestart (cpu, memory) policy for c1, RestartContainer (cpu, memory) for c2.
+   Verify that c1 is resized without restart, c2 is restarted on resize.
+1. NoRestart cpu, RestartContainer memory policy for c1. Resize c1 CPU only,
+   verify container is resized without restart.
+1. NoRestart cpu, RestartContainer memory policy for c1. Resize c1 memory only,
+   verify container is resized with restart.
+1. NoRestart cpu, RestartContainer memory policy for c1. Resize c1 CPU & memory,
+   verify container is resized with restart.
+
+### Negative Tests
+TBD
+
 ## Graduation Criteria
 
-TODO
+### Alpha
+- In-Place Pod Resouces Update functionality is implemented,
```

Comment:
Excending NodeRestriction with admitPodUpdate works, but NodeRestriction plugin is not enabled by default. Are there any concerns with default-enabling it as part of this KEP?

Reply:
> Are there any concerns with default-enabling it as part of this KEP?

Yes, do not change the default enablement. Cluster administrators that care about limiting a kubelet's ability to create pods will have it enabled.

Reply:
Ok, we'll have to document this and hope they read it closely enough -- it wont be out of the box security against a compromised node.

My initial feel is that NodeRestriction seems to work well though - the only bazel-test failure I'm seeing is because of my code limiting Node's access to this field.